lib/deaf_grandma.rb

```diff
@@ -13,15 +13,25 @@ def run!
 
     loop do
       user_input = get_user_input
-      p speak(user_input)
+      keeprunning, response = speak(user_input)
+      p response
+      exit if not keeprunning
     end
   end
 
 
   def speak(input)
+    @bye_counter += 1 if input == "BYE"
 
-    #Implement your code here <<<<<<<<<
+    if @bye_counter == 3
+      return false, "SEE YOU LATER SONNY!"
+    end
 
+    if input.upcase == input
+      return true, "NOT SINCE 1964!"
+    else
+      return true, "SPEAK UP SONNY!"
```

Comment:
These pairs of values are a code smell called [Data Clump](http://martinfowler.com/bliki/DataClump.html). 

Wanna try defining a [Struct](http://ruby-doc.org/core-2.2.0/Struct.html) for them to see if it improves things?